server/service/src/invoice/outbound_shipment/update/generate.rs

```diff
@@ -70,6 +83,43 @@ pub fn should_update_batches(
     }
 }
 
+// If status changed to allocated and above, remove unallocated lines
+fn unallocated_lines_to_trim(
+    connection: &StorageConnection,
+    invoice: &InvoiceRow,
+    status: &Option<UpdateOutboundShipmentStatus>,
+) -> Result<Option<Vec<InvoiceLineRow>>, RepositoryError> {
+    // Status sequence for outbound shipment: New, Allocated, Picked, Shipped
+    if invoice.status != InvoiceRowStatus::New {
+        return Ok(None);
+    }
+
+    let new_invoice_status = match UpdateOutboundShipmentStatus::full_status_option(status) {
+        Some(new_invoice_status) => new_invoice_status,
+        None => return Ok(None),
+    };
+
+    if new_invoice_status == InvoiceRowStatus::New {
+        return Ok(None);
+    }
+
+    // If new invoice status in not new and previous invoice status is new
```

Comment:
:nit - I think there is an 'in' which should be 'is'? was having trouble following the comment 🤷 
```suggestion
    // If new invoice status is not new and previous invoice status is new
```

Reply:
thanks